### PR TITLE
Feature/web hacks6 refactor

### PR DESCRIPTION
**Pull request by genki-aik**
### Description

Enter a short 1-3 sentence description about your PR.

Refactored UGAHacks-6 website that was created using create-react-app into Next.js and TypeScript. Moved CSS code to styles folder and moved static images into public folder. Removed NAV component as specified. 

### What are you changing?

* Refactored all React components into TypeScript and Next.js 
* Moved CSS files into Styles folder so that it is imported in one location, through _app.tsx
* Moved static images into public folder since Next.js can serve static files under folder called 'public'
* Removed NAV component as requested since scrolling functionality was not desired

### Did you increase testing or code coverage?

NO

### Did you add any depedencies to the project? 

YES, I added @types/react @types/react-dom for React typing.

### Did you update the relevant documentation?

NO, I am unsure which section to update on the documentation. I can look into updating Node.js, React, Typescript section or ugahacks-6 section.

### CODE REVIEW

File: web/ugahacks-6/tsconfig.json [hunk 1]
```diff
@@ -12,8 +12,9 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "jsx": "preserve"
+    "jsx": "preserve",
+    "strictNullChecks": true,
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "pages/components/HallOfFame/HallOfFame.tsx", "pages/components/HallOfFame/MediaCard.tsx", "pages/components/Sponsors/SponsorLogo.tsx"],
```
Comment (thatnealpatel) [inline web/ugahacks-6/tsconfig.json]: fix formatting e.g.

```json
"include": [
  "next-env.d.ts",
  "**/*.ts",
  ...
```

File: web/ugahacks-6/.gitignore [hunk 2]
```diff
@@ -32,3 +32,9 @@ yarn-error.log*
 
 # vercel
 .vercel
+
+#yarn
```
Comment (thatnealpatel) [inline web/ugahacks-6/.gitignore]: Why did you add this to the gitignore? You shouldn't be installing directly to a directory. You should be using the workspace syntax.

File: web/ugahacks-6/next.config.js [hunk 3]
```diff
@@ -1,12 +1,23 @@
 module.exports = {
   reactStrictMode: true,
-  async redirects(){
-    return[
+  env: {
+    HOSTNAME: "localhost",
```
Comment (thatnealpatel) [inline web/ugahacks-6/next.config.js]: I have a feeling this hardcoding is going to break in the pipelines and deployment.

**Review by thatnealpatel — CHANGES_REQUESTED** [review]
See comments.

**Comment by thatnealpatel** [comment]
Additionally, media URIs are not going to work. They do not follow the next.js convention. I'll let it slide for this PR due to how large the diff is but it'll have to be rectified in a similar manner to #27

File: web/ugahacks-6/next.config.js (hunk 3 above)
Comment (genki-aik) [reply web/ugahacks-6/next.config.js]: The environment variable HOSTNAME was a temporary fix to window element not found error located in serviceWorker.js. I can remove the environment variable but the window element not found error still needs to be addressed.

File: web/ugahacks-6/.gitignore (hunk 2 above)
Comment (genki-aik) [reply web/ugahacks-6/.gitignore]: Yes, I had a meeting with Andrei and started to use yarn workspace add <package>. Previously, I was incorrectly using yarn install and npm install directly.

**Review by thatnealpatel — APPROVED** [review]
